components/engine_rocks/src/misc.rs

```diff
@@ -130,7 +130,12 @@ impl MiscExt for RocksEngine {
         for cf in self.cf_names() {
             handles.push(util::get_cf_handle(self.as_inner(), cf)?);
         }
-        self.as_inner().flush_cfs(&handles, wait).map_err(r2e)
+        let latest_seqno = self.get_latest_sequence_number();
+        self.as_inner().flush_cfs(&handles, wait).map_err(r2e)?;
+        if let Some(listener) = self.flush_listener.as_ref() {
+            listener.notify_flush_cfs(latest_seqno);
```

Comment:
Why need to call it explicitly?

Reply:
Some CFs may have no data so we cannot get any notification from the listener.

Reply:
But if wait is false, the function is returned immediately. I think you need to check the seqno from manifest somehow.

Reply:
This function will return immediately if wait is false, `notify_flush_cfs ` cannot block. I don't think we could check manifest when CFs like default CF do not have any data, not any info will be recorded in manifest in this case.

Reply:
oh, so the notification should only be triggered when wait is `true`.

And for WAL, as I understood rocksdb actually traces WALs with corresponding memtables, everytime when `SwitchMemtable`, [a new WAL would be created](https://github.com/tikv/rocksdb/blob/ce0ea61aecd53ac02a0e7da2b166961c7d98650a/db/db_impl/db_impl_write.cc#L2181), and stale WALs would be created later when `FindObsoleteFiles` 

And the way rocksdb checks if a memtable was stale is through [log number](https://github.com/tikv/rocksdb/blob/ce0ea61aecd53ac02a0e7da2b166961c7d98650a/db/column_family.h#L587) recorded in manifest. This log number in manifest would be updated when [picking memtable for flushing](https://github.com/tikv/rocksdb/blob/ce0ea61aecd53ac02a0e7da2b166961c7d98650a/db/flush_job.cc#L189) through `VersionEdit`.

Reply:
Actually the reason we introduce the notification here is raft log GC, maybe we can just call `notify` when flushing memtables for raft log GC, instead of calling `notify` when every flushing?

Reply:
Removed this notification here, and this notification would be sent when we need flushing memtable to advance raft log GC. (will be included in next PRs)